include/zephyr/sys_clock.h

```diff
@@ -79,6 +79,36 @@ typedef struct {
  */
 #define K_TIMEOUT_EQ(a, b) ((a).ticks == (b).ticks)
 
+/** number of nsec per usec */
```

Comment:
maybe for a later update but these constants might be worth grouping in a `@name`d section. There are in fact probably several groups of such macros that could be grouped -- see below.
I know that you want to work on this incrementally but should you catch low-hanging fruits with regards to adding more structure where you can, please go for it :)

<img width="398" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/128251/90f6495d-32b1-411a-a7dd-acca2c78c785">